crates/matrix-crypto-ffi/src/machine.rs

```diff
@@ -517,7 +535,7 @@ impl OlmMachine {
             content: &'a RawValue,
         }
 
-        let event: SyncMessageEvent<RoomEncryptedEventContent> = serde_json::from_str(event)?;
+        let event: SyncMessageLikeEvent<RoomEncryptedEventContent> = serde_json::from_str(event)?;
         let room_id = Box::<RoomId>::try_from(room_id)?;
```

Comment:
This could also be made more readable now (maybe this was the case before already even):

```suggestion
        let room_id = RoomId::parse(room_id)?;
```

Reply:
All such instances were replaced in https://github.com/matrix-org/matrix-rust-sdk/pull/557/commits/a957a02d5de48f9a61b515e35a3d18a604e255fc.